src/context.sw

```diff
@@ -1,34 +1,27 @@
-library context;
 //! Functionality for accessing context-specific information about the current contract or message.
+library context;
+dep context/call_frames;
+dep context/registers;
 
 use ::contract_id::ContractId;
 use ::call_frames::*;
 
-dep context/call_frames;
-dep context/registers;
-
-/// Get the current contracts balance of coin `asset_id`
-pub fn this_balance(asset_id: ContractId) -> u64 {
-    let this_id = contract_id();
-    asm(balance, token: asset_id.value, contract_id: this_id.value) {
-        bal balance token contract_id;
+/// Retrieve the balance of asset 'asset_id' for the contract at 'contract_id'.
+pub fn balance(asset_id: ContractId, target: ContractId) -> u64 {
+    asm(balance, token: asset_id.value, id: target.value) {
+        bal balance token id;
```

Comment:
Would prefer a single `balance_of()` function (i.e. no `balance` and `balance_of_contract`), and implement `this_balance()` using `balance_of()`